### ISSUE TITLE
vllm.engine.async_llm_engine.AsyncEngineDeadError

### ISSUE THREAD

**Issue by SyedSherjeel** [vllm-project/vllm#1364]
Task finished unexpectedly. This should never happen! Please open an issue on Github. See stack trace above for the actual cause.
INFO 10-16 04:29:09 async_llm_engine.py:134] Aborted request cmpl-3c72dd29170e41ca93c2acbfabe2f326.

I can provide full stack trace of need be

**Comment by WoosukKwon** [comment]
Hi @SyedSherjeel, thanks for reporting the error. Could you provide the full stack trace and a reproducible script?

**Comment by SyedSherjeel** [comment]
Hi @WoosukKwon 
Thank you for response. Here is full stack trace. 

Traceback (most recent call last):
  File "uvloop/cbhandles.pyx", line 63, in uvloop.loop.Handle._run
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 37, in _raise_exception_on_finish
    raise exc
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 32, in _raise_exception_on_finish
    raise AsyncEngineDeadError(
vllm.engine.async_llm_engine.AsyncEngineDeadError: Task finished unexpectedly. This should never happen! Please open an issue on Github. See stack trace above for the actual cause.
INFO 10-17 05:30:01 async_llm_engine.py:134] Aborted request cmpl-9973fe4172d84e6a8a434239fd206f17.
Traceback (most recent call last):
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 28, in _raise_exception_on_finish
    task.result()
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 351, in run_engine_loop
    has_requests_in_progress = await self.engine_step()
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 330, in engine_step
    request_outputs = await self.engine.step_async()
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 191, in step_async
    output = await self._run_workers_async(
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 216, in _run_workers_async
    output = executor(*args, **kwargs)
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/worker/worker.py", line 319, in execute_model
    input_tokens, input_positions, input_metadata = self._prepare_inputs(
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/worker/worker.py", line 250, in _prepare_inputs
    tokens_tensor = torch.tensor(input_tokens,
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/torch/cuda/__init__.py", line 235, in _lazy_init
    raise RuntimeError(
RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "vibora/protocol/cprotocol.pyx", line 152, in vibora.protocol.cprotocol.Connection.handle_request
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/entrypoints/openai/api_server.py", line 291, in create_chat_completion
    async for res in result_generator:
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 436, in generate
    raise e
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 430, in generate
    async for request_output in stream:
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 70, in __anext__
    raise result
  File "uvloop/cbhandles.pyx", line 63, in uvloop.loop.Handle._run
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 37, in _raise_exception_on_finish
    raise exc
  File "/home/evobits/sherjeel/llm/env/lib/python3.8/site-packages/vllm/engine/async_llm_engine.py", line 32, in _raise_exception_on_finish
    raise AsyncEngineDeadError(
vllm.engine.async_llm_engine.AsyncEngineDeadError: Task finished unexpectedly. This should never happen! Please open an issue on Github. See stack trace above for the actual cause.
[2023-10-17T05:30:01+00:00] - 202.59.90.146 - "POST /v1/chat/completions" - 500 - PostmanRuntime/7.33.0

**Comment by SyedSherjeel** [comment]
So basically what I am trying to do is to replace fastapi and uvicorn with faster framework since its a bottleneck in vllm. Here are two framework I have attempted to replace it with same error.
https://github.com/vibora-io/vibora
https://github.com/sparckles/Robyn

**Comment by gesanqiu** [comment]
When update vllm to 0.2.1 and use tp>1, it will happen. I reproduce this issue by this command:
```shell
CUDA_VISIBLE_DEVICES=1,3 python /workdir/deploy/Inference_vLLM/vllm/entrypoints/openai/api_server.py --model /workdir/hf_models/models--BAAI--AquilaChat2-34B/ --served-model-name AquilaChat2-34B --trust-remote-code --tensor-parallel-size 2
```
But on vllm==0.2.0, it works well.

**Comment by WoosukKwon** [comment]
@gesanqiu Thanks for the confirmation. It should be fixed in `v0.2.1.post1` I uploaded today.

@SyedSherjeel Apologies for the bug.

**Comment by renillhuang** [comment]
Will this patch adding into mainline?
